querqy-core/src/main/java/querqy/rewrite/QueryRewriter.java

```diff
@@ -21,6 +22,19 @@ public interface QueryRewriter {
     
     Set<Term> EMPTY_GENERABLE_TERMS = Collections.emptySet();
 
-    ExpandedQuery rewrite(ExpandedQuery query);
+    RewrittenQuery rewrite(final ExpandedQuery query);
 
+    /**
+     * Rewrite the query. The caller of this method should expect that the query that was passed as an argument to this
+     * method could be modified.
+     *
+     * @param query The query to be rewritten
+     * @param searchEngineRequestAdapter Encapsulates the request context.
+     * @return The rewritten query.
+     *
+     */
+    default RewrittenQuery rewrite(final ExpandedQuery query,
+                                   final SearchEngineRequestAdapter searchEngineRequestAdapter) {
+        return rewrite(query);
```

Comment:
See above comment. No default implementation needed if we remove the rewrite(query) method.